wasp/src/main/java/com/orhanobut/wasp/VolleyImageHandler.java

```diff
@@ -0,0 +1,142 @@
+package com.orhanobut.wasp;
+
+import android.view.ViewGroup;
+import android.view.ViewTreeObserver;
+import android.widget.ImageView;
+
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.ImageLoader;
+import com.orhanobut.wasp.toolbox.WaspBitmapCache;
+
+/**
+ * @author Orhan Obut
+ */
+class VolleyImageHandler implements ImageHandler {
+
+    private static WaspBitmapCache bitmapCache;
+    private static ImageLoader imageLoader;
+
+    private WaspImage waspImage;
+
+    /**
+     * Current ImageContainer. (either in-flight or finished)
+     */
+    private ImageLoader.ImageContainer imageContainer;
+
+    @Override
+    public void init(WaspImage waspImage) {
+        this.waspImage = waspImage;
+        if (bitmapCache == null) {
+            bitmapCache = new WaspBitmapCache();
+        }
+        if (imageLoader == null) {
+            imageLoader = new ImageLoader(VolleyNetworkStack.getRequestQueue(), bitmapCache);
+        }
+    }
+
+    @Override
+    public void load() {
+        final ImageView imageView = waspImage.getImageView();
+        final ViewTreeObserver observer = imageView.getViewTreeObserver();
+        observer.addOnPreDrawListener(new ViewTreeObserver.OnPreDrawListener() {
+            @Override
+            public boolean onPreDraw() {
+                loadImage();
+                imageView.getViewTreeObserver().removeOnPreDrawListener(this);
+                return false;
+            }
+        });
+    }
+
+    private void loadImage() {
+        final ImageView imageView = waspImage.getImageView();
+        final String url = waspImage.getUrl();
+        final int defaultImage = waspImage.getDefaultImage();
+
+        int width = imageView.getMeasuredWidth();
+        int height = imageView.getMeasuredHeight();
+
+        boolean wrapWidth = false;
+        boolean wrapHeight = false;
+        if (imageView.getLayoutParams() != null) {
+            wrapWidth = imageView.getLayoutParams().width == ViewGroup.LayoutParams.WRAP_CONTENT;
+            wrapHeight = imageView.getLayoutParams().height == ViewGroup.LayoutParams.WRAP_CONTENT;
+        }
+
+        // if the view's bounds aren't known yet, and this is not a wrap-content/wrap-content
+        // view, hold off on loading the image.
+        boolean isFullyWrapContent = wrapWidth && wrapHeight;
+        if (width == 0 && height == 0 && !isFullyWrapContent) {
+            return;
+        }
+
+        // if there was an old request in this view, check if it needs to be canceled.
+        if (imageContainer != null && imageContainer.getRequestUrl() != null) {
+            if (imageContainer.getRequestUrl().equals(url)) {
```

Comment:
I would write like this:
<pre>
        String requestUrl;
        // if there was an old request in this view, check if it needs to be canceled.
        if (imageContainer != null && (requestUrl = imageContainer.getRequestUrl()) != null) {
            if (requestUrl.equals(url)) {
</pre>